pac4j-core/src/main/java/org/pac4j/core/io/Resource.java

```diff
@@ -0,0 +1,32 @@
+package org.pac4j.core.io;
+/**
+ * Interface for a resource descriptor that abstracts from the actual
+ * type of underlying resource, such as a file or class path resource.
+ *
+ * <p>An InputStream can be opened for every resource if it exists in
+ * physical form, but a URL or File handle can just be returned for
+ * certain resources. The actual behavior is implementation-specific.
+ *
+ */
+public interface Resource extends InputStreamSource {

```

Comment:
WriteableResource extends Resource so I don't think they should be merged for that reason. Resource is the common interface, input is for read and writeable is for write